Source/WebCore/dom/Document.cpp

```diff
@@ -3500,6 +3500,11 @@ void Document::setURL(const URL& url)
     if (SecurityOrigin::shouldIgnoreHost(m_url))
         m_url.setHostAndPort({ });
 
+    if (m_url.protocolIsBlob())
+        m_blobURLLifetimeExtension = m_url;
+    else
+        m_blobURLLifetimeExtension.clear();
+
```

Comment:
I wonder if we can / should make a URL container class or subclass, with an internal data member that automatically keeps the Blob URL alive if needed.

Something like:

```
class MightBeBlobURL {
public:
    MightBeBlobURL& operator=(URL& url) {
        m_url = url;
        if (m_url.protocolIsBlob())
            m_blobURLLifetimeExtension = m_url;
    }

    const URL& url() const { return m_url; }

private:
    URL m_url;
    BlobURLHandle m_blobURLLifetimeExtension;
};
```

Reply:
Yes, I think we likely should replace all BlobURLHandle instances with such a class. I can look into this in a follow-up if you don't mind as there are quite a few BlobURLHandle instances.

Reply:
Sure, makes sense.